install.sh

```
@@ -33,6 +22,761 @@ print_message "This script will install BirdNET-Go and its dependencies." "$YELL
+    
+    log_message "INFO" "Cleaning up old backups: $backup_count > $MAX_CONFIG_BACKUPS"
+    
+    # Remove oldest backups beyond the limit
+    local to_remove=$((backup_count - MAX_CONFIG_BACKUPS))
+    find "$CONFIG_DIR" -name "${CONFIG_BACKUP_PREFIX}*.yaml" -type f -print0 2>/dev/null | \
+        xargs -0 ls -t | tail -n "$to_remove" | while read -r old_backup; do
+            if rm -f "$old_backup" 2>/dev/null; then
+                log_message "INFO" "Removed old backup: $(basename "$old_backup")"
+                
+                # Remove from version tracking too
+                local backup_name=$(basename "$old_backup")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)